src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +44,29 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        AtomicInteger index = new AtomicInteger();
+        Map<Integer, Integer> mapsOfInt = numbers.stream()
+                .map(num -> index.getAndIncrement() + DETACH + num)
+                .collect(Collectors
+                        .toMap(e -> Integer.parseInt(String.valueOf(e.charAt(INDEX_FIRST))),
+                                e -> {
+                                    int value = Integer.parseInt(e.substring(
+                                            e.indexOf(":") + INCREMENT));
+                                    if (Integer.parseInt(
+                                            String.valueOf(
+                                                    e.charAt(INDEX_FIRST))) % NUM_CHECK_ODD
+                                                                            != NUM_REMAINDER) {
+                                        value--;
+                                    }
+                                    return value;
+                                }));
+        return mapsOfInt.values()
+                .stream()
+                .filter(e -> e % NUM_CHECK_ODD != NUM_REMAINDER).toList()
+                .stream()
```

Comment:
Why collect and then again to stream? :)
```suggestion
                .filter(e -> e % NUM_CHECK_ODD != NUM_REMAINDER)
```